File: Jim/src/sk/fiit/jim/garbage/plan/PlanTournamentGetUpFromBack.java
package sk.fiit.jim.garbage.plan;

/* HIGHSKILL NEEDED: 
 * 1 to pick a LOWSKILL
 */

import sk.fiit.jim.agent.AgentInfo;
import sk.fiit.jim.agent.highskill.*;
import sk.fiit.jim.agent.models.AgentModel;
import sk.fiit.jim.agent.models.EnvironmentModel;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 * Rewrited PlanTournamentGetUpFromBack from ruby
 * 
 * @author Adamik
 */

public class PlanTournamentGetUpFromBack extends Plan {

	
	public static PlanTournamentGetUpFromBack getInstance() {
		return new PlanTournamentGetUpFromBack();
	}

	boolean in_falled_pose 	= false;
	boolean up 				= false;

	public void replan() {
           AgentModel agentModel = AgentModel.getInstance();
           
		if(!beamed && !agentModel.isOnGround() && !in_falled_pose && !up && EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("Beam, wait for Fall");
			beamed 			= true;
			Vector3D start_position = Vector3D.cartesian(-5, 1, 0.4);
			Highskill_Queue.add(new Beam(start_position));
		}
		else if(beamed && !agentModel.isOnGround() && !in_falled_pose && !up && EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("Fall, wait for To falled pose");
			// falled 			= true;
			// @plan << LowSkill.new("fall_back")
		}
		else if(beamed && agentModel.isOnGround() && !in_falled_pose && !up && EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("To falled pose, after that I am ready for GetUp");
			in_falled_pose	= true;
			// @plan << LowSkill.new("rollback2")
		}
		else if(beamed && agentModel.isOnGround() && in_falled_pose && !up && !EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("GetUp, wait for Stand");
			up 				= true;
			Highskill_Queue.add(new GetUp());
		}
		else if(beamed && agentModel.isOnGround() && in_falled_pose && up && !EnvironmentModel.beamablePlayMode()) {
			AgentInfo.logState("Stand");
			// @plan << LowSkill.new("rollback")
		}
		else if(beamed && agentModel.isOnGround() && in_falled_pose && up && EnvironmentModel.beamablePlayMode()) {
			beamed 			= false;
			// falled 			= false;
			in_falled_pose 	= false;
			up 				= false;
		}
	}
}

File: Old_xml_and_scripts/Jim/plan/old-predUcesanim/planTournamentGetUpFromBack.rb
### planTournamentGetUpFromBack.rb ###
#	@author: A55Kickers				 #
######################################

class PlanTournamentGetUpFromBack < Plan

  def initialize
  	@beamed = false
  	@falled = false
  	@in_falled_pose = false
  	@up = false
  	super()
  end
  
  def replan
    @plan.clear
    @agentInfo.whereIsBall();
    @agentInfo.isBallMine();
    @agentInfo.nearBall();
    @agentInfo.ballControlPosition();
    @agentInfo = Java::sk.fiit.jim.agent.AgentInfo.getInstance
    @agentModel = Java::sk.fiit.jim.agent.models.AgentModel.getInstance
	
    if not @beamed and not @falled and not @in_falled_pose and not @up and EnvironmentModel.beamablePlayMode 
		@agentInfo.loguj('Beam, wait for Fall');
		@beamed = true
		@plan << Beam.new(Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(-3, 0, 0.4))
	elsif @beamed and not @falled and not @in_falled_pose and not @up and EnvironmentModel.beamablePlayMode 
		@agentInfo.loguj('Fall, wait for To falled pose');
		@falled = true
		@plan << LowSkill.new("padvzad")
	elsif @beamed and @falled and not @in_falled_pose and not @up and EnvironmentModel.beamablePlayMode 
		@agentInfo.loguj('To falled pose, after that I am ready for GetUp');
		@in_falled_pose = true
		@plan << LowSkill.new("rollback2")
    elsif @beamed and @falled and @in_falled_pose and not @up and not EnvironmentModel.beamablePlayMode 
    	@agentInfo.loguj('GetUp, wait for Stand');
    	@up = true
     	@plan << GetUp.new(Proc.new{me.on_ground?})
    elsif @beamed and @falled and @in_falled_pose and @up and not EnvironmentModel.beamablePlayMode
     	@agentInfo.loguj('Stand');
     	@plan << LowSkill.new("rollback")
    elsif @beamed and @falled and @in_falled_pose and @up and EnvironmentModel.beamablePlayMode
    	@beamed = false
  		@falled = false
  		@in_falled_pose = false
  		@up = false
    end
  end
end